core/src/test/java/org/apache/iceberg/TestSnapshotRefParser.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Objects;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestSnapshotRefParser {
```

Comment:
would it make sense to add some tests where:
* the json string is null/empty
* the json string contains field names that it doesn't understand
* the json string omits required fields, such as the `snapshot-id` or the `type`
* certain fields contain values that can't be parsed, such as `type` having some garbage string or `snapshot-id` having a string that can't be parsed to a long

So far we're only testing the happy path, but not the cases where things could go wrong

Reply:
Updated to include these cases. For the unknown field names, currently in the parsing logic we would simply ignore them as we are selecting the desired fields from the JSON node. I believe this should be the desired implementation but let me know if you feel otherwise.